packages/wonder-blocks-popover/src/components/popover-keypress-listener.js

```diff
@@ -1,25 +1,51 @@
 // @flow
 
 import * as React from "react";
+import ReactDOM from "react-dom";
+
+import typeof PopoverContent from "./popover-content.js";
+import typeof PopoverContentCore from "./popover-content-core.js";
 
 type Props = {|
     /**
      * Called when `esc` is pressed
      */
     onClose: () => mixed,
+
+    /**
+     * Popover Content ref.
+     * Will close the popover when clicking outside this element.
+     */
+    contentRef?: RefObject<PopoverContentCore | PopoverContent>,
+|};
+
+type State = {|
+    /**
+     * Tracks the first click triggered by the click event listener.
+     */
+    isFirstClick: boolean,
 |};
 
 /**
  * A component that, when mounted, calls `onClose` when Escape is pressed.
  * @see @khanacademy/wonder-blocks-modal/components/modal-launcher.js
  */
-export default class PopoverKeypressListener extends React.Component<Props> {
+export default class PopoverKeypressListener extends React.Component<
+    Props,
+    State,
+> {
+    state: State = {
+        isFirstClick: true,
+    };
+
     componentDidMount() {
         window.addEventListener("keyup", this._handleKeyup);
+        window.addEventListener("click", this._handleClick);
```

Comment:
It would be good to add a unit test verifying that `onClose` is called if we press `click` outside the popover ref.